jwala-services/src/main/java/com/cerner/jwala/service/app/impl/ApplicationServiceImpl.java

```diff
@@ -128,7 +132,42 @@ public Application getApplication(final String name) {
     public Application updateApplication(UpdateApplicationRequest updateApplicationRequest, User anUpdatingUser) {
         updateApplicationRequest.validate();
 
-        return applicationPersistenceService.updateApplication(updateApplicationRequest);
+        final Application application = applicationPersistenceService.updateApplication(updateApplicationRequest);
+        updateApplicationWarMetaData(updateApplicationRequest, application);
+        return application;
+    }
+
+    private void updateApplicationWarMetaData(UpdateApplicationRequest updateApplicationRequest, Application application) {
+        final String appWarName = application.getWarName();
+        if (!appWarName.isEmpty()) {
+            final String appName = application.getName();
+            try {
+                String originalJsonMetaData = groupPersistenceService.getGroupAppResourceTemplateMetaData(application.getGroup().getName(), appWarName);
+                ResourceTemplateMetaData originalMetaData = resourceService.getMetaData(originalJsonMetaData);
+                ResourceTemplateMetaData updateMetaData = new ResourceTemplateMetaData(
+                        originalMetaData.getTemplateName(),
+                        originalMetaData.getContentType(),
+                        originalMetaData.getDeployFileName(),
+                        originalMetaData.getDeployPath(),
+                        originalMetaData.getEntity(),
+                        updateApplicationRequest.isUnpackWar(),
```

Comment:
What about other meta data updates, does it mean I can only update unpackwar flag?